### PR TITLE
Changed: RunCommandService Refactored

### PR DESCRIPTION
**Pull request by rubinstory**
Changed: RunCommandService Refactored

### CODE REVIEW

File: app/src/main/java/com/termux/app/RunCommandService.java [hunk 1]
```diff
@@ -71,125 +72,63 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
-        // If EXTRA_RUNNER is passed, use that, otherwise check EXTRA_BACKGROUND and default to Runner.TERMINAL_SESSION
-        executionCommand.runner = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_RUNNER,
-            (intent.getBooleanExtra(RUN_COMMAND_SERVICE.EXTRA_BACKGROUND, false) ? Runner.APP_SHELL.getName() : Runner.TERMINAL_SESSION.getName()));
-        if (Runner.runnerOf(executionCommand.runner) == null) {
-            errmsg = this.getString(R.string.error_run_command_service_invalid_execution_command_runner, executionCommand.runner);
-            executionCommand.setStateFailed(Errno.ERRNO_FAILED.getCode(), errmsg);
-            TermuxPluginUtils.processPluginExecutionCommandError(this, LOG_TAG, executionCommand, false);
-            return stopService();
-        }
+        Integer invalidRunner = checkRunner(intent, executionCommand);
```
Comment (agnostic-apollo) [inline app/src/main/java/com/termux/app/RunCommandService.java:86]: Could just return `false` from function and do

`if(!checkRunner()) return stopService()`

**Comment by agnostic-apollo** [comment]
Also the `RunCommandService` is gonna be refactored and split soon for new API design, so above changes will be lost.

**Comment by rubinstory** [comment]
@agnostic-apollo 
We didn't know that this file will be refactored and splitted, so we'll just close this PR.

**Comment by agnostic-apollo** [comment]
Apart from `checkInvalidAction()` extraction, these were reasonable changes, although sequence of function definitions should have been the same as they were called. Split will indeed remove some stuff, specially intent extraction and creation, I currently haven't decided on final design, but already have a local implementation.

**Pull request closed without merging**